2003-02-04  Jakub Jelinek  <jakub@redhat.com>

	* dwarf2out.c (dwarf2out_finish): Add AT_comp_dir
	attribute even if input file name is absolute, but one of the
	includes is relative.

--- gcc-3.3.1/gcc/dwarf2out.c.dwarf2-AT_comp_dir	2003-07-16 11:41:16.000000000 +0200
+++ gcc-3.3.1/gcc/dwarf2out.c	2003-07-16 11:59:16.000000000 +0200
@@ -12585,6 +12585,22 @@ dwarf2out_finish (input_filename)
   limbo_die_node *node, *next_node;
   dw_die_ref die = 0;
 
+  if (get_AT (comp_unit_die, DW_AT_comp_dir) == NULL)
+    {
+      char *wd = getpwd ();
+      unsigned i;
+
+      if (wd != NULL)
+	{
+	  for (i = 1; i < file_table.in_use; i++)
+	    if (file_table.table[i][0] != DIR_SEPARATOR)
+	      {
+		add_AT_string (comp_unit_die, DW_AT_comp_dir, wd);
+		break;
+	      }
+	}
+    }
+
   /* Traverse the limbo die list, and add parent/child links.  The only
      dies without parents that should be here are concrete instances of
      inline functions, and the comp_unit_die.  We can ignore the comp_unit_die.
